### PR TITLE
md-tangle: init at 1.9.1

### PR DESCRIPTION
**Pull request by michaelhthomas**
###### Motivation for this change
This adds [md-tangle](https://github.com/joakimmj/md-tangle) to nixpkgs, a python tool which "tangles" source code from Markdown documents, similar to Emacs Org-mode and org-babel-tangle.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -32274,4 +32274,6 @@ in
   zthrottle = callPackage ../tools/misc/zthrottle { };
 
   zktree = callPackage ../applications/misc/zktree {};
+
+  md-tangle = callPackage ../development/tools/md-tangle { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:32278]: Please sort this entry.

File: pkgs/development/tools/md-tangle/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  propagatedBuildInputs = [
+    setuptools
+  ];
+
+  meta = with lib; {
+    description = "Generates (\"tangles\") source code from Markdown documents";
+    homepage = "https://github.com/joakimmj/md-tangle";
+    license = licenses.mit;
+    # maintainers = with maintainers; [ michaelhthomas ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/md-tangle/default.nix:25]: Please add yourself as a maintainer by adding an alphabetically sorted entry to `maintainers/maintainer-list.nix` and commit this change into a separate commit with the message ``maintainers: add NAME`` into this PR.

Comment (michaelhthomas) [reply pkgs/development/tools/md-tangle/default.nix]: Done, thanks for the help!

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (michaelhthomas) [reply pkgs/top-level/all-packages.nix]: Done!

File: pkgs/development/tools/md-tangle/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    owner = "joakimmj";
+    repo = "md-tangle";
+    rev = "v${version}";
+    sha256 = "1xsqbq7kbgq5r050b07ixfn5v0w61a8nwd9d0vbbyjqgc8i6m6c6";
+  };
+
+  propagatedBuildInputs = [
+    setuptools
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/tools/md-tangle/default.nix:20]: checkInputs and pythonImportsCheck?

**Comment by AndersonTorres** [comment]
I will catch this.

**Pull request closed without merging**